azurelinuxagent/pa/provision/default.py

```diff
@@ -57,45 +58,68 @@ def run(self):
         if not conf.get_provision_enabled():
             logger.info("Provisioning is disabled, skipping.")
             self.write_provisioned()
-            self.report_ready()
+            logger.info("signaling...")
+            if self.report_ready():
+                self.write_signaled()
+                logger.info("signaling complete")
+            else:
+                logger.info("signaling incomplete")
```

Comment:
This block of code appears three times, and in all three cases subsequent code does not care which branch of the `if` was taken. You can hoist this up into a function local to the run() method.